src/app/examples/form/components/form.component.spec.ts

```diff
@@ -1,35 +1,104 @@
+import { initialState } from '../form.reducer';
+import { ActionFormUpdate } from '../form.actions';
 
-describe('FormComponent', () => {
-  let component: FormComponent;
-  let fixture: ComponentFixture<FormComponent>;
+fdescribe('FormComponent', () => {
   let store: MockStore<State>;
+  let renderResult: RenderResult;
+  let dispatchSpy: jasmine.Spy;
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      imports: [TestingModule],
-      declarations: [FormComponent],
-      providers: [NotificationService]
-    }).compileComponents();
+  beforeEach(async () => {
+    renderResult = await createComponent<FormComponent>(
+      '<anms-form></anms-form>',
+      {
+        declarations: [FormComponent],
+        imports: [TestingModule],
+        providers: [NotificationService],
+        detectChanges: false
+      }
+    );
 
     store = TestBed.get(Store);
     store.setState(createState(initialState));
-    fixture = TestBed.createComponent(FormComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  }));
+    dispatchSpy = spyOn(store, 'dispatch');
+
+    renderResult.fixture.detectChanges();
+  });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should submit when form is valid', async () => {
+    renderResult.input(
+      renderResult.getByLabelText('anms.examples.form.placeholder1'),
+      {
+        target: {
+          value: '@tim_deschryver'
+        }
+      }
+    );
+    renderResult.input(
+      renderResult.getByLabelText('anms.examples.form.placeholder2'),
+      {
+        target: {
+          value: 'mysuperawesomeandsecurepassword'
+        }
+      }
+    );
+    renderResult.input(
+      renderResult.getByLabelText('anms.examples.form.placeholder3'),
+      {
+        target: {
+          value: 'foo@bar.baz'
+        }
+      }
+    );
+    renderResult.input(
+      renderResult.getByLabelText('anms.examples.form.placeholder5'),
+      {
+        target: {
+          value: '1991-12-31'
+        }
+      }
+    );
+    renderResult.input(
+      renderResult.getByLabelText('anms.examples.form.placeholder4'),
+      {
+        target: {
+          value: 'no description needed here'
+        }
+      }
+    );
+    renderResult.click(renderResult.getByLabelText('anms.examples.form.text1'));
+    renderResult.click(renderResult.getByText('anms.examples.form.send'));
+    const { type, payload } = new ActionFormUpdate({
+      form: {
+        autosave: false,
+        username: '@tim_deschryver',
+        password: 'mysuperawesomeandsecurepassword',
+        email: 'foo@bar.baz',
+        description: 'no description needed here',
+        requestGift: true,
+        birthday: new Date(1991, 12, 31, 0, 0, 0),
+        rating: 0
+      }
+    });
+    const { birthday, ...formValue } = payload.form;
+    expect({ ...dispatchSpy.calls.mostRecent().args[0] }).toEqual({
```

Comment:
If someone got a cleaner way of doing this, feel free to chip in 😅 
Problems I encountered:
- JS Object vs TS Class
- Birthday is a date object, this was giving some troubles; hence the check if it's filled in, instead of comparing both values.